torch/_export/trace.py

```diff
@@ -0,0 +1,212 @@
+import contextlib
+import copy
+from typing import Callable, Tuple, Generator, Dict, Optional
+from unittest.mock import patch
+import dataclasses
+
+import torch
+import weakref
+import torch._dynamo as torchdynamo
+from torch._decomp import core_aten_decompositions
+from torch._dispatch.python import enable_python_dispatcher
+from torch.nn.utils import stateless
+from torch.utils import _pytree as pytree
+from torch.fx.experimental.symbolic_shapes import StrictMinMaxConstraint
+from torch.utils._sympy.value_ranges import ValueRanges
+import sympy
+
+from torch._functorch.aot_autograd import (
+    AOTConfig,
+    create_aot_dispatcher_function,
+    default_partition,
+    run_functionalized_fw_and_collect_metadata,
+)
+
+from torch.fx.experimental.proxy_tensor import (
+    get_proxy_slot,
+    get_torch_dispatch_modes,
+    has_proxy_slot,
+    make_fx,
+    ProxyTorchDispatchMode,
+    set_proxy_slot,
+)
+
+from torch._functorch.eager_transforms import _unwrap_all_tensors_from_functional
+
+from .workflow import ExportedProgram
+
+CORE_ATEN_DECOMPOSITIONS_TABLE = core_aten_decompositions()
+
+__all__ = ["do_not_use_experimental_export"]
+
+
+def _aot_capture(mod, flat_args):
```

Comment:
Does this change have anything to do with inline constraints? Not sure why this is in the same diff, maybe I'm missing something?

Reply:
It has to do with circular imports 😅 

I need to add torch/_export/constraints.py to dynamo's skipfiles so that dynamo knows to trace through these functions. But there was a circular import error in functorch's aot_autograd tests, something like: aot_autograd tests import dynamo --> dynamo imports torch/_export/constraints.py in its skipfiles --> torch/_export/__init__.py imports aot_autograd.

So if I move tracer code to another file (not __init__.py) it'll solve that issue 😋